src/librustc_mir/transform/simplify.rs

```diff
@@ -124,7 +124,7 @@ impl<'a, 'tcx> CfgSimplifier<'a, 'tcx> {
                 }
 
                 let data = &mut self.basic_blocks[bb];
-                data.statements.extend(new_stmts);
+                data.statements.extend(new_stmts.drain(..));
```

Comment:
I feel like maybe `merge_successor` count take `bb` and modify `self.basic_blocks[bb].statements`, draining from the original block directly, without any of these copies into a buffer? But that's neither here nor there.

Reply:
Fixed, I thought that would be more difficult since there is a risk that the side effect of appending affects something else.  Was pretty trivial though when I actually looked at it.